src/promptflow/promptflow/_utils/multimedia_utils.py

```diff
@@ -224,12 +222,50 @@ def load_multimedia_data(inputs: Dict[str, FlowInputDefinition], line_inputs: di
             error_type_and_message = f"({ex.__class__.__name__}) {ex}"
             raise LoadMultimediaDataError(
                 message_format="Failed to load image for input '{key}': {error_type_and_message}",
-                key=key, error_type_and_message=error_type_and_message,
-                target=ex.target
+                key=key,
+                error_type_and_message=error_type_and_message,
+                target=ex.target,
             ) from ex
     return updated_inputs
 
 
+def convert_multimedia_path_to_url_recursively(value: Any, convert_path_to_url: Callable):
+    process_func = partial(_convert_multimedia_path_to_url, converter=convert_path_to_url)
+    return _process_multimedia_dict_recursively(value, process_func=process_func)
+
+
+# For converter, the input path may be relative path or absolute path.
+def _convert_multimedia_path_to_url(image_dict: dict, converter: Callable[[str], str]):
+    """Convert image path to url in image dict"""
+    if is_multimedia_dict(image_dict):
+        for key in image_dict:
+            format, resource = _get_multimedia_info(key)
+            if resource == "path":
+                path = image_dict[key]
+                url = converter(path)
+                return {f"data:image/{format};url": url}
+    return image_dict
+
+
+def convert_multimedia_url_to_path_recursively(value: Any, convert_url_to_path: Callable[[str], str]):
```

Comment:
A question: why not also define this convert_url_to_path in utils?

Reply:
That should be a real client sdk implementation. Such as using Compute Instance Identity to download url.
So, couldn't define in there.